Updated Feature with content from Whats-new-10-0

Vic-Cooper opened the pull request:
Transferred feature details and new features from whats-new-10-0.

# **Please read;**
## **PR Workflow for the Graphics repository:**
* **All PRs must be opened as draft initially**
* Reviewers can be added while the PR is still in draft
* The PR can be marked as “Ready for Review” once the reviewers have confirmed that **no more changes are needed**
* Tests will start automatically after the PR is marked as “Ready for Review”
* **Do not use [skip ci]** - this can break some of our tooling
* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status
Describe what manual/automated tests were performed for this PR

---
### Comments to reviewers
Notes for the reviewers you have assigned.

JordanL8 commented:
@sebastienlagarde is this everything we need to move from the 10.x what's new to the features page?

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -175,6 +183,12 @@ HDRP provides a screen-space solution for reflection and refraction. It uses the
 
 HDRP provides a screen-space solution for distortion. Distortion is similar to refraction except that it is purely artistic driven. For more information, see the [distortion](Surface-Type.md#TransparencyInputs) documentation.
 
+### IES Profiles and light cookies
```
This should be moved under #light section

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -246,7 +260,7 @@ HDRP uses real-world physical light units, so you can easily light your Scene in
+### Ray tracing (Preview)
 
 HDRP uses ray tracing to replace some of its screen space effects, shadowing techniques, and Mesh rendering techniques.
 Real time raytracing effect are currently in Preview and behavior could change in the future.
```
We are missing raytrace subsurface scattering in this list.

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -295,6 +309,18 @@ HDRP uses a physically-based Camera system that works seamlessly with the other
+
+### Custom Pass API
+
+The Custom Pass API allows you to render GameObjects from another point of view, like a disabled camera, within the rendering of your main Camera. This API also comes with built-in support for rendering Depth, Normal and Tangent into an RTHandle.
```
this API shouldnt be a separate section but should be inside ### Custom Passes section following the other description.

sebastienlagarde reviewed (commented):
Hey, so having a quick look, 

we are missing
- In debug mode we should add "Quad Overdraw and Vertex Density", "lighting debug mode", Light layer and volume debug mode..
- Have we added in exposure section the histogram? 
- In MSAA section we should add that we support alpha to mask
- in screen space reflection we should add that we support SSR on transparent material
- to double check but decal section should mention decal layer
- Path tracing section should mention voluemtric , fog and reference depth of field
- We need to add somewhere accumulation motion blur

Maybe we could add a section named 'recorder' which is the tool done by unity and link to it and say that we have accumulation motion blur, AOV export, progressive path tracing, reference depth of field compatible with it?

- In planar reflection section we should add that we have accurate filtering for planar reflection (close to reference)

- in light section let's add that we have guildelines / icon to help setup physical based light value. Maybe with a screenshots?

this is what I spotted on a short run :)

thanks!

Vic-Cooper commented:
I've addressed all comments and added missing sections. Please could you both review the following: 
@sebastienlagarde - Please check that I haven't missed anything In my edits!
@JordanL8 - Please could you look at the images I added for Path Tracing? some of them cropped better than others. I think I could do with a banner image for Exposure too but I'm not sure where to find one. If you could cast an eye over my formatting too that would also be helpful :)

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -167,14 +167,43 @@ For more information, including the full list of light properties, see the [Ligh
+
+![](Images/PlanarReflectionFiltering-Feature.png)
+
+Planar reflection probe filtering combines the result of planar reflection and surfaces smoothness. HDRP includes accurate filtering for planar reflection that is close to being physically-based.
```
Let's reword it like this: (no need for "filtering" )

We will remove planar reflection from the previous section:

HDRP supports cubemap reflection taking into account surface smoothness to help you produce realistic reflections in your Scene in real time. For more information, see the [introduction to Reflection Probes](Reflection-Probes-Intro.md)

 ### Planar reflection probe

=> HDRP supports Planar reflection probe taking into account surfaces smoothness, allowing to mimic mirror to rough floor. HDRP includes accurate filtering for planar reflection that is close to being physically-based.

sebastienlagarde replied:
and I add a link to this pages: https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@10.0/manual/Planar-Reflection-Probe.html

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -167,14 +167,43 @@ For more information, including the full list of light properties, see the [Ligh
+
+### Screen-space global illumination
+
+![](Images/HDRPFeatures-SSGI.png)
```
move this one just after ### Screen-space reflection and refraction

sebastienlagarde replied:
Remove 

"as a fallback for ray-traced global illumination (RTGI)". It is not appropriate here.

sebastienlagarde replied:
and thus removed 
Since this feature is a fallback for RTGI, for more information, see [Ray-traced Global Illumination](Ray-Traced-Global-Illumination.md).

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -167,14 +167,43 @@ For more information, including the full list of light properties, see the [Ligh
+
+![](Images/HDRPFeatures-IESProfiles.png)
+
+HDRP supports the Illuminating Engineering Society's (IES) file format for describing the distribution of light from a light source. HDRP supports the IES profile for Point, Spot (Cone, Pyramid, and Box), and rectangular Area [Lights](Light-Component.md). You can also mix the IES profile with [cookies](https://docs.unity3d.com/Manual/Cookies.html) and even use the profile and cookie mix for [ligh ...
```
Move this section under the Light types section a bit above

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -167,14 +167,43 @@ For more information, including the full list of light properties, see the [Ligh
+
+HDRP allows you to calculate exposure using several methods. This includes: 
+
+- Histogram exposure: HDRP's exposure implementation computes a histogram of the image which allows you to select high and low percentile values to discard. Discarding outlying values in the shadows or highlights helps to calculate a more stable exposure.
```
Ask @FrancescoC-unity for a bannar pictures for this one) thanks

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -272,18 +302,52 @@ HDRP includes its own purpose-built implementation for post-processing to produc
 
 ![Posterize](Images/HDRPFeatures-CustomPostProcess.png)
```
Note: let's move custom postprocessing just before custom pass which is below

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -272,18 +302,52 @@ HDRP includes its own purpose-built implementation for post-processing to produc
-HDRP allows you to add your own custom post processes integrated with the volume framework. They can be injected after opaque and sky object, before builtin post processes or after builtin post processes. For more information, see the [Custom Post-processing documentation](Custom-Post-Process.md).
+HDRP allows you to add your own custom post processes integrated with the volume framework. They can be injected after opaque and sky object, before the [temporal anti-aliasing](Anti-Aliasing.md#TAA) pass, before builtin post processes or after builtin post processes. For more information, see the [Custom Post-processing documentation](Custom-Post-Process.md).
+
+### Path tracing
```
Move path tracing (and all stuff inside it) just before the tools section

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -272,18 +302,52 @@ HDRP includes its own purpose-built implementation for post-processing to produc
+
+![](Images/Path_tracing_recording-Feature.png)
+
+HDRP includes a recording API which you can use to render effects such as high-quality accumulation motion blur and converged path-traced images. These techniques create the final "converged" frame by combining information from multiple intermediate sub-frames. The new API allows your scripts to extract the properly converged final frames and perform further processing or save them to disk.
```
We need to separte Accumulation motion blur  and path tracer convergence APIs.

Let's move Accumulation motion blur APIs. in Camerea section just after postprocess and 

let ath tracer convergence APIs here

for accumulation you have:
HDRP includes a recording API which you can use to render effects such as high-quality accumulation motion blur. The technique create the final frame by combining information from multiple intermediate sub-frames. The new API allows your scripts to extract the properly final frames and perform further processing or save them to disk.

for pathtracer you have:

HDRP includes a recording API which you can use to render converged path-traced images. The techniques create the final "converged" frame by combining information from multiple intermediate sub-frames. The new API allows your scripts to extract the properly converged final frames and perform further processing or save them to disk.

sebastienlagarde commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -311,6 +385,45 @@ The Render Pipeline Debugger contains many debugging and visualization tools to
 
+### Debug modes
+
+HDRP includes debug modes that can help you to set the correct exposure for your Scene.
```
=> HDRP includes debug modes that can help you to debug your assets, your lighting, your scene.

sebastienlagarde reviewed (commented):
Thanks for all the change, just a few stuff remaining and some stuff to move

JordanL8 commented on com.unity.render-pipelines.high-definition/Documentation~/HDRP-Features.md:
```diff
@@ -272,18 +302,52 @@ HDRP includes its own purpose-built implementation for post-processing to produc
+
+#### Path-traced depth of field
+
+![](Images/Path-traced-DOF-Feature.png)
```
Strangely, if I go to View file in the top right, this image or the one below (  ![](Images/Path_tracing_recording-Feature.png)  ) aren't showing up. I can't immediately see what's up but it might be a git sync thing. If you can see them in Typora (or whatever you're using) though then it's fine :)

Vic-Cooper replied:
Checked this with you and it does appear to be fine :)

Vic-Cooper commented:
Updates to the physically-based lighting UI isn't yet implemented. I'll request to be added to the PR and add these features when they're live.

sebastienlagarde commented:
Not sur to understand what you mean by physical light unit isn't available yet. the PR is merged. let's sync on slack. Will merge this PR in the mean time. THanks for all the update!
